### PR TITLE
Fix Calls to count()

### PR DESCRIPTION
**Pull request by bbatsche**
### Description

Add logic and checks to avoid passing `null` to `count()` and make sure that any classes with a public `count()` method implement `Countable` interface.

### Fixed Issues (if relevant)

1. magento-engcom/php-7.2-support#7: Warning: count(): Parameter must be an array or an object that implements Countable

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Review by buskamuza — APPROVED** [review]

**Comment by buskamuza** [comment]
There are still failures with `count()`.
See:
* https://travis-ci.org/magento-engcom/php-7.2-support/jobs/355658287
* https://travis-ci.org/magento-engcom/php-7.2-support/jobs/355658288
* https://travis-ci.org/magento-engcom/php-7.2-support/jobs/355658291
* https://travis-ci.org/magento-engcom/php-7.2-support/jobs/355658292
* https://travis-ci.org/magento-engcom/php-7.2-support/jobs/355658293

**Review by buskamuza — CHANGES_REQUESTED** [review]
Please run builds and fix issues with `count()`

File: dev/tests/integration/framework/tests/unit/testsuite/Magento/Test/Bootstrap/EnvironmentTest.php [hunk 1]
```diff
@@ -9,6 +9,7 @@
  */
 namespace Magento\Test\Bootstrap;
 
+ob_start();
```
Comment (bbatsche) [inline dev/tests/integration/framework/tests/unit/testsuite/Magento/Test/Bootstrap/EnvironmentTest.php]: @joanhe Is there a reason for this? I feel like it definitely does not belong here

Comment (joanhe) [reply dev/tests/integration/framework/tests/unit/testsuite/Magento/Test/Bootstrap/EnvironmentTest.php]: @bbatsche This is to resolve test failure in php 7.2. 
```session_id(): Cannot change session id when headers already sent in ```

Comment (bbatsche) [reply dev/tests/integration/framework/tests/unit/testsuite/Magento/Test/Bootstrap/EnvironmentTest.php]: Maybe try adding this right above the call that's failing, or somewhere in one of the `setUp*()` methods. No code or function calls should be in this file and not inside of the class itself.

Comment (joanhe) [reply dev/tests/integration/framework/tests/unit/testsuite/Magento/Test/Bootstrap/EnvironmentTest.php]: This is the only place works.

**Review by buskamuza — APPROVED** [review]

**Pull request merged**